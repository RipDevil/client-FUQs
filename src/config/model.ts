import { createStore, createEvent } from 'effector';
import defaultConfig from './defaultConfig';

export interface ConfigType {
  server: string;
  api: {
    fuq?: {
      get?: string;
      put?: string;
      post?: string;
      delete?: string;
    };
  };
}

const $config = createStore<ConfigType>(defaultConfig);

const configUpdate = createEvent<ConfigType>('update config');

$config.on(
  configUpdate,
  (state, payload): ConfigType => ({ ...state, ...payload })
);

export { configUpdate, $config };
