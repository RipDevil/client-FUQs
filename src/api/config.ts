import { AxiosError } from 'axios';
import { useQuery } from 'react-query';

import { call } from './lib';
import { ConfigType } from 'config/model';

// type Params = {
//   [key: string]: string
// };

/**
 * Fetches the config file with basename
 */
export const useConfig = () => {
  var a = 'Test';

  return useQuery<unknown, AxiosError, ConfigType>("Get configuration file", () => call('/config.json', 'GET'));
};
