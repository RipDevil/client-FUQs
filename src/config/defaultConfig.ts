const defaultConfig = {
  server: '',
  api: {
    fuq: {
      get: "/fuq",
      put: "/fuq",
      post: "/fuq/:id",
      delete: "/fuq/:id"
    }
  },
};

export default defaultConfig;