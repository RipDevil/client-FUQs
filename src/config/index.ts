export * from './wrapper';
export * from './model';
