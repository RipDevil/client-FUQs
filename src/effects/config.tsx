import { createEvent } from 'effector';

const configUpdate = createEvent<Object>('update config');

export {
  configUpdate
};