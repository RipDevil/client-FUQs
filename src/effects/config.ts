import { createEvent } from 'effector';
import { ConfigType } from 'stores/config';
const configUpdate = createEvent<ConfigType>('update config');

export { configUpdate };
