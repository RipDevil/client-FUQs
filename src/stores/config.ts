import { createStore } from 'effector';

import { configUpdate } from 'effects/config';

export interface ConfigType {
  server: string;
  routes: {
    fuq?: string;
    badmin?: string;
    root?: string;
  };
}

const $config = createStore<ConfigType>({ server: '', routes: {} });

$config.on(configUpdate, (_, payload): ConfigType => payload);

export default $config;
