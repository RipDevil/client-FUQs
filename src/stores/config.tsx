import { createStore } from 'effector';

import { configUpdate } from 'effects/config';

const $config = createStore<Object>({});

$config
  .on(configUpdate, (_, payload) => payload);

export default $config;