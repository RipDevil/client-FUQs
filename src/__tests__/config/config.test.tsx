import { QueryClient, QueryClientProvider } from 'react-query';
import { render, waitFor, cleanup } from '@testing-library/react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import { ConfigWrapper } from 'config';

let queryClient: any;
let mockAxios: any;

beforeAll(() => {
  queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
        retry: false,
      },
    },
  });
  mockAxios = new MockAdapter(axios);
});

afterEach(() => {
  mockAxios.reset();
  cleanup();
});

describe('Config wrapper tests', () => {
  it('Should render child elements', async () => {
    mockAxios.onGet('/config.json').reply(200, {
      server: 'test',
    });

    const { getByText } = render(
      <QueryClientProvider client={queryClient}>
        <ConfigWrapper>{'test'}</ConfigWrapper>
      </QueryClientProvider>,
    );

    const loadingElement = getByText(/Loading config/i);
    expect(loadingElement).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();

    const resolvedElement = await waitFor(() => getByText(/test/i));
    expect(resolvedElement).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });

  it('Should render nothing on get config error', async () => {
    mockAxios.onGet('/config.json').reply(404);

    render(
      <QueryClientProvider client={queryClient}>
        <ConfigWrapper>{'test'}</ConfigWrapper>
      </QueryClientProvider>,
    );

    await waitFor(() => document.body.querySelector('div')?.innerHTML === '');
    expect(document.body).toMatchSnapshot();
  });
});
