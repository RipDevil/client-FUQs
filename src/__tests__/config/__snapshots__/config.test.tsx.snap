// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config wrapper tests Should render child elements 1`] = `
.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 1000;
  margin: -50px 0px 0px -50px;
  max-width: 140px;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  background-color: rgba(253,253,253,1);
}

.c2 {
  text-align: center;
  margin-top: 1rem;
}

<body>
  <div>
    <div
      class="c0"
    />
    <div
      class="c1"
    >
      <div>
        <i
          class="fas fa-10x fa-spin fa-cog"
        />
      </div>
      <div
        class="c2"
      >
        <h3
          class="ant-typography ant-typography-danger"
        >
          Loading config
        </h3>
      </div>
    </div>
  </div>
</body>
`;

exports[`Config wrapper tests Should render child elements 2`] = `
<body>
  <div>
    <div>
      test
    </div>
  </div>
</body>
`;

exports[`Config wrapper tests Should render nothing on get config error 1`] = `
<body>
  <div />
</body>
`;
